Lib/ntpath.py

```diff
@@ -122,54 +122,169 @@ def join(path, *paths):
 # colon) and the path specification.
 # It is always true that drivespec + pathspec == p
 def splitdrive(p):
-    """Split a pathname into drive/UNC sharepoint and relative path specifiers.
-    Returns a 2-tuple (drive_or_unc, path); either part may be empty.
-
-    If you assign
-        result = splitdrive(p)
-    It is always true that:
-        result[0] + result[1] == p
-
-    If the path contained a drive letter, drive_or_unc will contain everything
-    up to and including the colon.  e.g. splitdrive("c:/dir") returns ("c:", "/dir")
-
-    If the path contained a UNC path, the drive_or_unc will contain the host name
-    and share up to but not including the fourth directory separator character.
-    e.g. splitdrive("//host/computer/dir") returns ("//host/computer", "/dir")
-
-    Paths cannot contain both a drive letter and a UNC path.
-
+    """Split path p conservatively into a drive and remaining path.
+    Returns a 2-tuple, (drive, rest). Either component may be empty.
+
+    If the source path contains a DOS drive (i.e. a letter plus a colon), the
+    remaining path is everything after the colon.
+
+    DOS drive examples:
+
+        splitdrive('C:') == ('C:', '')
+        splitdrive('C:dir') == ('C:', 'dir')
+        splitdrive('C:/') == ('C:', '/')
+        splitdrive('C:/dir') == ('C:', '/dir')
+
+    A UNC path is parsed as follows:
+
+        drive
+        vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
+        "//" domain "/" junction ["/" junction] ["/" object]
+                                                ^^^^^^^^^^^^
+                                                rest
+
+    The UNC root must be exactly two separators. Other separators may be
+    repeated.
```

Comment:
Thank you for picking this up, but, fair warning, it's going to need more work. If you're up for that, great. I'd love to get your opinions on what needs to be done.

Steve Dower introduced the idea of leveraging `PathCchSkipRoot()` in Windows, and I like that idea, despite some reservations on my part. So I think `splitdrive()` should try to conform with `PathCchSkipRoot()`. As such, repetition of separators between the domain and junctions (e.g. UNC server and share) should be parsed as empty values for those components. Similar behavior would be extended to "\\\\?\\UNC" paths. 

At the time I wrote this, I was thinking to support whatever paths work in practice, but on closer scrutiny even `GetFullPathNameW()` handles the initial slashes for the server and share components without normalizing repeated slashes. For example:

```
>>> n = GetFullPathNameW('//server//file', len(buf), buf, byref(filepart))
>>> print(buf.value)
\\server\file
>>> print(filepart.value)
file

>>> n = GetFullPathNameW('////file', len(buf), buf, byref(filepart))
>>> print(buf.value)
\\\file
>>> print(filepart.value)
file
```
For normal UNC filepaths, if `os.chdir()` handles the path as a UNC path instead of as a rooted path on the current drive, then `splitdrive()` should split out the 'drive' component, even if it's malformed (e.g. empty server or share component). Extend this behavior to "\\\\?\\UNC" paths, as `PathCchSkipRoot()` does, even though they're not valid for the current working directory.